easy_my_coop_connector/components/subscription_request_adapter.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2020 Coop IT Easy SCRL fs
+#   Robin Keunen <robin@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from werkzeug.exceptions import BadRequest
+
+from odoo import _
+from odoo.exceptions import UserError, ValidationError
+from odoo.fields import Date
+
+from .abstract_emc_adapter import AbstractEMCAdapter
+
+
+class SubscriptionRequestAdapter(AbstractEMCAdapter):
+    _model = "subscription.request"
+    _service = "subscription-request"
+
+    def search(self, date_from=None, date_to=None):
+        url = self._get_url()
+
+        params = {}
+        if date_from:
+            params.update({"date_from": Date.to_string(date_from)})
+        if date_to:
+            params.update({"date_to": Date.to_string(date_to)})
+
+        sr_list = self.backend.http_get_content(url, params=params)
+        return {
+            "count": sr_list["count"],
+            "rows": [self.to_write_values(row) for row in sr_list["rows"]],
+        }
+
+    def update(self, data):
+        if self.record is None:
+            raise AttributeError("record field must be set for an update.")
+        external_id = self.record.binding_id.external_id
+        url = "/".join((self._get_url(), str(external_id), "update"))
+        try:
+            request_dict = self.backend.http_post_content(url, data)
```

Comment:
The PUT method is implemented by `base_rest` module but not patch. Even though - as you explained - PATCH is probably the best VERB to use to upload partial resources, I'll use the PUT method because the API schema is documented in the code AND by an apispec interface:
<img width="409" alt="Screenshot 2022-01-03 at 17 23 51" src="https://user-images.githubusercontent.com/3082119/147954964-d6bcce43-2bdc-40d1-83b8-34cb7d8c8a65.png">
<img width="375" alt="Screenshot 2022-01-03 at 17 24 15" src="https://user-images.githubusercontent.com/3082119/147954973-a6be13ef-7658-419f-87f4-250ac86ab3e1.png">